stingray/lightcurve.py

```diff
@@ -123,8 +140,9 @@ def __add__(self, other):
                                  "of same dimension and equal.")
 
```

Comment:
By default, I would not "apply GTIs", in the sense of eliminating counts from the light curve.
There are procedures where the fact of being outside GTIs is not a big deal. For example, we might have an interval where the spectral information is bad and we have a high rate of rejected events because of an improper status of the detectors. In this case, this is very bad for aperiodic variability, that will be completely altered, or for spectral studies. But if we, say, are looking for pulsations, we might want to use all events we have regardless of the fact that the time interval is bad. 
So, by default, we declare that such things are GTIs exist for our light curve. Different methods will take this into account or not depending on the possible impact of bad time intervals or gaps in the data.
In the case of this function, we are adding two light curves as we did before. Then, we cross the two GTIs and declare in the new light curve the time intervals that are reliable, while saving the full information also for intervals that are not.